lib/elixir/lib/inspect.ex

```diff
@@ -69,15 +69,13 @@ defimpl Inspect, for: Atom do
   defp color_key(nil), do: :nil
   defp color_key(_), do: :atom
 
-  def inspect(false), do: "false"
-  def inspect(true), do: "true"
-  def inspect(nil), do: "nil"
-
   def inspect(atom) do
     binary = Atom.to_string(atom)
 
-    cond do
-      valid_alias?(binary) ->
+    case classify(atom) do
+      :boolean_or_nil ->
+        binary
+      :alias ->
         # If the given binary is just a succession like "Elixir.Elixir.Elixir",
         # inspect it as is, otherwise, strip the leading "Elixir".
         if only_elixir?(binary) do
```

Comment:
I have just realized this code is also wrong. For example, `Elixir.Elixir.Atom` is being inspected as `Elixir.Atom` which is then interpreted as `Atom`. Instead of `only_elixir?(binary)`, we need to check if the first two segments is "Elixir", if so, we keep "Elixir", otherwise we remove it.